cineast-core/src/main/java/org/vitrivr/cineast/core/features/abstracts/AbstractSegmentExporter.java

```diff
@@ -0,0 +1,124 @@
+package org.vitrivr.cineast.core.features.abstracts;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.vitrivr.cineast.core.data.segments.SegmentContainer;
+import org.vitrivr.cineast.core.db.PersistencyWriterSupplier;
+import org.vitrivr.cineast.core.db.setup.EntityCreator;
+import org.vitrivr.cineast.core.features.extractor.Extractor;
+import org.vitrivr.cineast.core.util.LogHelper;
+
+abstract public class AbstractSegmentExporter implements Extractor {
+
+  protected static final Logger LOGGER = LogManager.getLogger();
+
+  private static final String PROPERTY_NAME_DESTINATION = "destination";
+
+  /**
+   * Destination path for the audio-segment.
+   */
+  private final Path destination;
+
+  protected String fileExtension;
+
+  protected String dataUrlPrefix;
+
+  /**
+   * Default constructor
+   */
+  public AbstractSegmentExporter() {
+    this(new HashMap<>());
+  }
+
+  /**
+   * Default constructor. The AudioSegmentExport can be configured via named properties in the provided HashMap.
+   * <p>
+   * Supported parameters:
+   *
+   * <ol>
+   *      <li>destination: Path where files should be stored.</li>
+   * </ol>
+   *
+   * @param properties HashMap containing named properties
+   */
+  public AbstractSegmentExporter(HashMap<String, String> properties) {
+    this.destination = Path.of(properties.getOrDefault(PROPERTY_NAME_DESTINATION, "."));
```

Comment:
I am not convinced that `.` is a sensible default here.
My suggestion would be `$(pwd)/export` to not fill the current working directory with individual files?

Reply:
I agree with this and am fixing this. Just for the record, originally AudioSegmentExporter did use '.'